tests/e2e/multicluster/multicluster_multiprimary_test.go

```diff
@@ -0,0 +1,479 @@
+	SetDefaultEventuallyTimeout(180 * time.Second)
+	SetDefaultEventuallyPollingInterval(time.Second)
+
+	// debugInfoLogged := false
+
+	BeforeAll(func(ctx SpecContext) {
+		if !skipDeploy {
+			// Deploy the Sail Operator on both clusters
+			Expect(kubectl.CreateNamespace(namespace, kubeconfig)).To(Succeed(), "Namespace failed to be created on Primary Cluster")
+			Expect(kubectl.CreateNamespace(namespace, kubeconfig2)).To(Succeed(), "Namespace failed to be created on Remote Cluster")
+
+			Expect(helm.Install("sail-operator", filepath.Join(project.RootDir, "chart"), "--namespace "+namespace, "--set=image="+image, "--kubeconfig "+kubeconfig)).
+				To(Succeed(), "Operator failed to be deployed in Primary Cluster")
+
+			Eventually(common.GetObject).
+				WithArguments(ctx, clPrimary, kube.Key(deploymentName, namespace), &appsv1.Deployment{}).
+				Should(HaveCondition(appsv1.DeploymentAvailable, metav1.ConditionTrue), "Error getting Istio CRD")
+			Success("Operator is deployed in the Primary namespace and Running")
+
+			Expect(helm.Install("sail-operator", filepath.Join(project.RootDir, "chart"), "--namespace "+namespace, "--set=image="+image, "--kubeconfig "+kubeconfig2)).
+				To(Succeed(), "Operator failed to be deployed in Remote Cluster")
+
+			Eventually(common.GetObject).
+				WithArguments(ctx, clRemote, kube.Key(deploymentName, namespace), &appsv1.Deployment{}).
+				Should(HaveCondition(appsv1.DeploymentAvailable, metav1.ConditionTrue), "Error getting Istio CRD")
+			Success("Operator is deployed in the Remote namespace and Running")
+		}
+	})
+
+	Describe("Multi-Primary Multi-Network configuration", func() {
+		// Test the Multi-Primary Multi-Network configuration for each supported Istio version
+		for _, version := range supportedversion.List {
+			Context("Istio version is: "+version.Version, func() {
+				When("Istio resources are created in both clusters with multicluster configuration", func() {
+					BeforeAll(func(ctx SpecContext) {
+						pushIntermediateCA()
+
+						multiclusterYAML := `
+apiVersion: sailoperator.io/v1alpha1
+kind: Istio
+metadata:
+  name: default
+spec:
+  version: %s
+  namespace: %s
+  values:
+    global:
+      meshID: %s
+      multiCluster:
+        clusterName: %s
+      network: %s`
+						multiclusterPrimaryYAML := fmt.Sprintf(multiclusterYAML, version.Name, controlPlaneNamespace, "mesh1", "cluster1", "network1")
+						Log("Istio CR Primary: ", multiclusterPrimaryYAML)
+						Expect(kubectl.CreateFromString(multiclusterPrimaryYAML, kubeconfig)).To(Succeed(), "Istio Resource creation failed on Primary Cluster")
+
+						multiclusterRemoteYAML := fmt.Sprintf(multiclusterYAML, version.Name, controlPlaneNamespace, "mesh1", "cluster2", "network2")
+						Log("Istio CR Remote: ", multiclusterRemoteYAML)
+						Expect(kubectl.CreateFromString(multiclusterRemoteYAML, kubeconfig2)).To(Succeed(), "Istio Resource creation failed on Remote Cluster")
```

Comment:
nit: again since this is a multi-primary deployment, this is not a `Remote` cluster.

Reply:
Changed